bakery/pipeline.js

```diff
@@ -0,0 +1,104 @@
+const yaml = require('js-yaml')
+
+const taskLookUpBook = require('./tasks/look-up-book')
+const taskFetchBook = require('./tasks/fetch-book')
+const taskAssembleBook = require('./tasks/assemble-book')
+const taskBakeBook = require('./tasks/bake-book')
+const taskMathifyBook = require('./tasks/mathify-book')
+const taskBuildPdf = require('./tasks/build-pdf')
+
+// FIXME: This mapping should be in the COPS resource
+const Status = Object.freeze({
+  QUEUED: 1,
+  ASSIGNED: 2,
+  PROCESSING: 3,
+  FAILED: 4,
+  SUCCEEDED: 5
+})
+
+const reportToOutputProducer = (status, extras) => {
+  return {
+    put: 'output-producer',
+    params: {
+      id: 'output-producer/id',
+      status_id: status,
+      ...extras
+    }
+  }
+}
+
+const resourceTypes = [
+  {
+    name: 'output-producer',
+    type: 'docker-image',
+    source: {
+      repository: 'openstax/output-producer-resource'
+    }
+  }
+]
+
+const resources = [
+  {
+    name: 'cnx-recipes',
+    type: 'git',
+    source: {
+      uri: 'https://github.com/openstax/cnx-recipes.git'
+    }
+  },
+  {
+    name: 'output-producer',
+    type: 'output-producer',
+    source: {
+      api_root: '((pdf-job-queue-url))',
+      status_id: 1
+    }
+  },
+  {
+    name: 's3',
+    type: 's3',
+    source: {
+      bucket: 'ce-pdf-spike',
+      access_key_id: '((aws-sandbox-secret-key-id))',
+      secret_access_key: '((aws-sandbox-secret-access-key))',
+      skip_download: true
+    }
+  }
+]
+
+const bakeryJob = {
+  name: 'bakery',
+  plan: [
+    { get: 'output-producer', trigger: true, version: 'every' },
+    reportToOutputProducer(Status.ASSIGNED),
+    { get: 'cnx-recipes' },
+    taskLookUpBook,
+    reportToOutputProducer(Status.PROCESSING),
+    taskFetchBook,
+    taskAssembleBook,
+    taskBakeBook,
+    taskMathifyBook,
+    taskBuildPdf,
+    {
+      put: 's3',
+      params: {
+        file: 'artifacts/*.pdf',
+        acl: 'public-read',
+        content_type: 'application/pdf'
+      }
+    }
+  ],
+  on_success: reportToOutputProducer(Status.SUCCEEDED, {
+    pdf_url: 'book/pdf_url'
+  }),
+  on_failure: reportToOutputProducer(Status.FAILED),
+  on_error: reportToOutputProducer(Status.FAILED),
+  on_abort: reportToOutputProducer(Status.FAILED)
+}
+
+const config = {
+  resource_types: resourceTypes,
+  resources: resources,
+  jobs: [bakeryJob]
+}
+
+console.log(yaml.safeDump(config))
```

Comment:
Can you include a comment at the top of the `.yml` file indicating that it was autogenerated and instructions for re-generating?